accounts/static/accounts/js/onboarding_movies.js

```diff
@@ -0,0 +1,522 @@
+      movies_watched = extractIds(data.movies_watched);
+      watchlist = extractIds(data.watchlist);
+      movies_rewatch = extractIds(data.movies_rewatch);
+      movies_blocked = extractIds(data.movies_excluded);
+    });
+}
+
+function extractIds(movieList) {
+  return new Set(movieList.map((movie) => movie.pk));
+}
+
+// Random movie fetch when scrolling past halfway point
+document.addEventListener("DOMContentLoaded", function () {
+  movieContainer.addEventListener("scroll", () => {
+    searchString = searchbar.value.trim();
+    threshold = (movieContainer.scrollHeight * 3) / 4;
+    if (!searchString && movieContainer.scrollTop >= threshold) {
+      console.log("Fetching more movies...");
+      fetchMovies();
+    }
+  });
+});
+
+// Hides the tooltip when the user clicks outside of it
+document.addEventListener("click", function (event) {
+  if (currentTooltiptext && !currentTooltiptext.contains(event.target)) {
+    currentTooltiptext.style.display = "none";
+  }
+});
+
+/************************************
+ *
+ * User preference button functions
+ *
+ ************************************/
+
+/**
+ * This function attempts to update the liked status of a movie.
+ * @param {String} id the id of the movie to be updated
+ */
+function addLiked(id) {
+  liked_button = document.getElementById(id + "_upvote");
+  intID = parseInt(id);
+  if (movies_liked.has(intID)) {
+    liked_button.style.backgroundColor = tooltipBackgroundColor;
+    movies_liked.delete(intID);
+
+    return;
+  } else if (movies_disliked.has(intID)) {
+    disliked_button = document.getElementById(id + "_dislike");
+    disliked_button.style.backgroundColor = tooltipBackgroundColor;
+    movies_disliked.delete(intID);
+  } else {
+    watched_button = document.getElementById(id + "_seen");
+    watched_button.style.backgroundColor = "blue";
+    movies_watched.add(intID);
+  }
+
+  movies_liked.add(intID);
+  liked_button.style.backgroundColor = likeGreen;
+}
+
+/**
+ * This function attempts to update the dislike status of a movie.
+ * @param {String} id the id of the movie to be updated
+ */
+function addDisliked(id) {
+  dislike_button = document.getElementById(id + "_dislike");
+  intID = parseInt(id);
+
+  if (movies_disliked.has(intID)) {
+    dislike_button.style.backgroundColor = tooltipBackgroundColor;
+    movies_disliked.delete(intID);
+
+    return;
+  } else if (movies_liked.has(intID)) {
+    liked_button = document.getElementById(id + "_upvote");
+    liked_button.style.backgroundColor = tooltipBackgroundColor;
+    movies_liked.delete(intID);
+  } else {
+    watched_button = document.getElementById(id + "_seen");
+    watched_button.style.backgroundColor = "blue";
+    movies_watched.add(intID);
+  }
+
+  movies_disliked.add(intID);
+  dislike_button.style.backgroundColor = "red";
+}
+
+/**
+ * This function attempts to update the watched status of a movie.
+ * @param {String} id the id of the movie to be updated
+ */
+function addSeen(id) {
```

Comment:
these all seem to have very similar purposes and there's a lot of redundant code but whatevs, it's saturday at 10am